poetry/console/commands/shell.py

```diff
@@ -0,0 +1,51 @@
+from os import environ
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        # This variable will be set if the current shell was spawned by poetry
+        current_venv = environ.get("POETRY_ACTIVE")
```

Comment:
I think it'd be better if you provide a default in case there is no `POETRY_ACTIVE` variable in environ --> `environ.get("POETRY_ACTIVE", None)`

Reply:
I believe `None` is already the default fallback value for `.get`

Reply:
Sorry, my bad :) You're right (didn't know that, always used it explicitly)

Reply:
Maybe you're right in doing so! :)